test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -643,6 +644,71 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Offlined CPU API", func() {
+		It("[disruptive] should set offline cpus after deploy PAO", func() {
+			By("Backing up the profile")
+			initialProfile := profile.DeepCopy()
+
+			for _, node := range workerRTNodes {
+				onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUInt, err := strconv.Atoi(onlineCPUCount)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(onlineCPUInt).Should(BeNumerically(">=", 3))
+			}
+
+			// Create new performance with offlined
+			reserved := performancev2.CPUSet("0")
+			isolated := performancev2.CPUSet("1")
+			offlined := performancev2.CPUSet("2-3")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable UserLevelNetworking and add Devices in Profile")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reserved,
+				Isolated: &isolated,
+				Offlined: &offlined,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			//Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
+				Eventually(func() bool {
+					offlinedOutput, err := nodes.ExecCommandOnNode([]string{"cat", "/sys/devices/system/cpu/offline"}, &node)
+					Expect(err).ToNot(HaveOccurred())
+					offlinedCPUSet, err := cpuset.Parse(offlinedOutput)
+					offlinedCPUSetProfile, err := cpuset.Parse(string(offlined))
+					Expect(err).ToNot(HaveOccurred())
+					return offlinedCPUSet.Equals(offlinedCPUSetProfile)
+				}, cluster.ComputeTestTimeout(15*time.Minute, false), 1*time.Minute).Should(BeTrue(), "could not get correct offline cpus")
+			}
+
+			By("Reverting the Profile")
```

Comment:
this should probably moved (in some form) in a `AfterEach` or `defer` block (I'm fine with either), to make sure it runs even if an `Expect` in lines 683-693 fails. We totally don't want to pollute the cluster state

Reply:
So right. 
Add beforeEach and afterEach to make it clearer and less buggy

Reply:
thanks, looks nice